src/async_block_store.rs

```diff
@@ -145,7 +150,59 @@ impl<U: Unblocker> AsyncBlockStore<U> {
         f: impl FnOnce(&mut BlockStore) -> T + Send + 'static,
     ) -> BoxFuture<T> {
         let inner = self.inner.clone();
-        self.unblocker
+        self.runtime
             .unblock(move || f(DerefMut::deref_mut(&mut inner.lock().unwrap())))
     }
+
+    /// A gc loop that runs incremental gc in regular intervals
+    ///
+    /// Gc will run as long as this future is polled.
+    pub async fn gc_loop(self, config: GcConfig) -> crate::Result<()> {
+        loop {
+            debug!("gc_loop running incremental gc");
+            self.incremental_gc(config.min_blocks, config.target_duration)
+                .await?;
+            self.runtime.sleep(config.interval / 2).await;
+            debug!("gc_loop running incremental delete orphaned");
+            self.incremental_delete_orphaned(config.min_blocks, config.target_duration)
+                .await?;
+            self.runtime.sleep(config.interval / 2).await;
+        }
+    }
+}
+
+/// Configuration for the gc loop
+///
+/// This is done as a config struct since we might have additional parameters here in the future,
+/// such as limits at which to do a full gc.
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct GcConfig {
+    /// interval at which gc runs
+    ///
+    /// note that this is implemented as delays between gcs, so it will not run exactly
+    /// at this interval, but there will be some drift if gc takes long.
+    pub interval: Duration,
+
+    /// minimum number of blocks to collect in any case
+    ///
+    /// Using this parameter, it is possible to guarantee a minimum rate with which gc will be
+    /// able to keep up. It is min_blocks / interval.
+    pub min_blocks: usize,
+
+    /// The target maximum gc duration of a single gc.
+    ///
+    /// This can not be guaranteed, since we guarantee to collect at least `min_blocks`.
+    /// But as soon as this duration is exceeded, the incremental gc will stop doing additional
+    /// work.
+    pub target_duration: Duration,
```

Comment:
`max_duration`? You probably didn't take that name because it can't be guaranteed, but still ..

Reply:
I got max duration in the other methods. Decided that it is not a good name, exactly because it can not be guaranteed but is just a hint to the gc about how long to take.